Support for OS X 10.13?

Hello! I am not familiar with C++ but was wondering if it would be feasible to make changes for Manifold to be compatible with Mac OS X 10.13 (and possibly lower.)

I'm building with `make -k` on OS X 10.13 and the two errors below in `csg_tree.cpp` (lines 85 and 87) appear to be the only blockers to a full build:

`manifold/src/manifold/src/csg_tree.cpp:85:14: error: call to unavailable function 'get': introduced in macOS 10.14`
`     return std::get<std::shared_ptr<const Manifold::Impl>>(p).get();`
`manifold/src/manifold/src/csg_tree.cpp:87:14: error: call to unavailable function 'get': introduced in macOS 10.14`
`     return std::get<std::shared_ptr<Manifold::Impl>>(p).get();`

Naïve searching turned up a StackOverflow answer suggesting that there were [doable workarounds](https://stackoverflow.com/questions/52521388/stdvariantget-does-not-compile-with-apple-llvm-10-0) using `std::get_if` and some boilerplate error handling code, but again I'm not versed in C++ and am unsure of the real amount of work to ensure compatibility.

Thank you for your work on this library, its inclusion in OpenSCAD has made my workflow much faster and easier :)

--- a/bindings/python/manifold3d.cpp
+++ b/bindings/python/manifold3d.cpp
@@ -322,8 +322,7 @@ NB_MODULE(manifold3d, m) {
              std::optional<nb::ndarray<uint32_t, nb::shape<3>>> &normalIdx) {
             glm::ivec3 v(0);
             if (normalIdx.has_value()) {
-              if (normalIdx->ndim() != 1 ||
-                  normalIdx->shape(0) != 3)
+              if (normalIdx->ndim() != 1 || normalIdx->shape(0) != 3)
                 throw std::runtime_error("Invalid vector shape, expected (3)");
               auto value = *normalIdx;
               v = glm::ivec3(value(0), value(1), value(2));

--- a/bindings/python/manifold3d.cpp
+++ b/bindings/python/manifold3d.cpp
@@ -322,10 +322,10 @@ NB_MODULE(manifold3d, m) {
              std::optional<nb::ndarray<uint32_t, nb::shape<3>>> &normalIdx) {
             glm::ivec3 v(0);
             if (normalIdx.has_value()) {
-              if (normalIdx.value().ndim() != 1 ||
-                  normalIdx.value().shape(0) != 3)
+              if (normalIdx->ndim() != 1 ||
+                  normalIdx->shape(0) != 3)
                 throw std::runtime_error("Invalid vector shape, expected (3)");
-              auto value = normalIdx.value();
+              auto value = *normalIdx;
               v = glm::ivec3(value(0), value(1), value(2));
             }
             return self.GetMeshGL(v);
@@ -624,7 +624,7 @@ NB_MODULE(manifold3d, m) {
                                                      runOriginalID->size());
 
             if (runTransform.has_value()) {
-              auto runTransform1 = runTransform.value();
+              auto runTransform1 = *runTransform;
               if (runTransform1.ndim() != 3 || runTransform1.shape(1) != 4 ||
                   runTransform1.shape(2) != 3)
                 throw std::runtime_error(
@@ -637,7 +637,7 @@ NB_MODULE(manifold3d, m) {
               out.faceID = toVector<uint32_t>(faceID->data(), faceID->size());
 
             if (halfedgeTangent.has_value()) {
-              auto halfedgeTangent1 = halfedgeTangent.value();
+              auto halfedgeTangent1 = *halfedgeTangent;
               if (halfedgeTangent1.ndim() != 3 ||
                   halfedgeTangent1.shape(1) != 3 ||
                   halfedgeTangent1.shape(2) != 4)

--- a/src/manifold/src/csg_tree.cpp
+++ b/src/manifold/src/csg_tree.cpp
@@ -82,9 +82,9 @@ using SharedImpl = std::variant<std::shared_ptr<const Manifold::Impl>,
 struct GetImplPtr {
   const Manifold::Impl *operator()(const SharedImpl &p) {
     if (std::holds_alternative<std::shared_ptr<const Manifold::Impl>>(p)) {
-      return std::get<std::shared_ptr<const Manifold::Impl>>(p).get();
+      return std::get_if<std::shared_ptr<const Manifold::Impl>>(&p)->get();
     } else {
-      return std::get<std::shared_ptr<Manifold::Impl>>(p).get();
+      return std::get_if<std::shared_ptr<Manifold::Impl>>(&p)->get();
     }
   };
 };
@@ -495,7 +495,7 @@ std::shared_ptr<Manifold::Impl> CsgOpNode::BatchBoolean(
     group.run_and_wait(process);
     SharedImpl r;
     queue.try_pop(r);
-    return std::get<std::shared_ptr<Manifold::Impl>>(r);
+    return *std::get_if<std::shared_ptr<Manifold::Impl>>(&r);
   }
 #endif
   // apply boolean operations starting from smaller meshes

